src/arrayMethodSort.js

```diff
@@ -6,6 +6,41 @@
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
     // write code here
+
+    if (compareFunction) {
+      for (let x = 0; x < this.length - 1; x++) {
+        for (let i = 0; i < this.length - 1; i++) {
+          const a = this[i];
+          const b = this[i + 1];
+
+          if (compareFunction.bind(this)(a, b) > 0) {
+            this[i] = b;
+            this[i + 1] = a;
+          };
+
+          if ((compareFunction.bind(this)(a, b) <= 0)) {
```

Comment:
This if condition is redundant, as the values are already set correctly in the previous if condition.